src/modules/fstab/main.py

```diff
@@ -151,23 +151,23 @@ def generate_crypttab_line_info(self, partition):
         if not mapper_name or not luks_uuid:
             return None
 
-        password = "/crypto_keyfile.bin"
         crypttab_options = self.crypttab_options
+        if os.path.isfile(os.path.join(self.root_mount_point, "crypto_keyfile.bin")):
+            password = "/crypto_keyfile.bin"
+        else:
+            password = "none"
 
-        # Set crypttab password for partition to none and remove crypttab options
         # if root partition was not encrypted
-        if any([p["mountPoint"] == "/"
-                and "luksMapperName" not in p
-                for p in self.partitions]):
-            password = "none"
+        if any([p["mountPoint"] == "/" and
+               "luksMapperName" not in p for p in self.partitions]):
```

Comment:
I think it would be worth adding a helper function 

```
def is_unencrypted(partition):
    return "luksMapperName" not in partition

def is_partition_encrypted(mountpoint : str, partitions):
    return any(p["mountPoint"] == mountpoint and is_unencrypted(p) for p in self.partitions)
```

Reply:
It sure would be nice but it won't reduce amount of code or simplify it.
In the future (when systemd-boot is the standard or grub gains full LUKS2 support) we might be able to drop 1/2 of the code.

Also, testing fstab module is a real pain.
Full installation for each possible setup is required after each change.
That's why my initial PR was so "ugly" but minimal.

If some is volunteers to testing, I am all for it.